### PR TITLE
Orchardcore testing

### LINKED ISSUE #11865: OrchardCore Testing
### Is your feature request related to a problem? Please describe.
When developing an Orchard Core module/app it would be nice to have a Testing Library to get you started with unit and integration testing in OrchardCore terms

### Describe the solution you'd like
Create an OrchardCore.Testing library by extracting reusable code from OrchardCore.Tests

### PR DESCRIPTION
**Pull request by MichaelPetrinolis**
Extract reusable testing assets to an OrchardCore.Testing library

closes #11865

### CODE REVIEW

File: src/docs/topics/testing/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Unit and Integration Testing with Orchard Core
+
+When developing an OrchardCore solutions, most of the times you want to unit test the most mandatory parts of your application/module.
+That's why we provide a testing library, that will allow you to get started easily. We use the same library for out own OrchardCore.Tests
```
Comment (Skrypt) [inline src/docs/topics/testing/README.md:4]: "our own" instead of "out own"

File: src/docs/topics/testing/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+## How use the library
+
+The best way to see how it works, is by check the OrchardCore.Tests project in the source code.
+
+For integration testing, you create create a SiteContext object and configure it by using the SiteContextConfig static object.
```
Comment (Skrypt) [inline src/docs/topics/testing/README.md:10]: "create create" twice

**Comment by MichaelPetrinolis** [comment]
@Skrypt I wrote the readme as a placeholder ;) - if the PR is to be accepted, we should enrich it with some guidelines and examples on how to use it for unit/integration testing.

**Comment by hyzx86** [comment]
Great !!
Maybe we can add some workflow related tests,

Test code for liquid / jint,   such as JSON serialization

**Comment by hishamco** [comment]
Thanks for your initiative @MichaelPetrinolis, this could be closed - if you don't mind - in favour of #13018

**Comment by MichaelPetrinolis** [comment]
> Thanks for your initiative @MichaelPetrinolis, this could be closed - if you don't mind - in favour of #13018

Sure, no problem, I paused this PR anyway, I used it in my projects but as long as there is the #13018 , I will adjust my code when this gets merged :) Thank you for taking over.

**Pull request closed without merging**